manifests/master.pp

```diff
@@ -1,243 +1,132 @@
-# Type: Boolean
-# Default: true
-#
-# If true, log to the local system logger over UDP port 514.
-#
-# [*syslog_facility*]
-# Type: String
-# Default: 'LOCAL6'
-#
-# The syslog facility to which to report if using syslog.
-#
-# [*syslog_message_format*]
-# Type: String
-# Default: '%logger[%thread]: %msg'
-#
-# The Logback compatible syslog message format. For more information, see the
-# Logback documentation for 'SuffixPattern'.
-#
-# [*log_level*]
-# Type: One of ['TRACE','DEBUG','INFO','WARN','ERROR','OFF']
-# Default: 'WARN'
-#
-# A syslog severity string limiting the messages reported. Be aware that
-# anything above 'WARN' will provide a massive amount of logs at each puppet
-# run.
-#
-# == Authors
-#
-# * Trevor Vaughan <tvaughan@onyxpoint.com>
+# @author Trevor Vaughan <tvaughan@onyxpoint.com>
 #
 class pupmod::master (
-  $bind_address          = '0.0.0.0',
-  $ca_bind_address       = '0.0.0.0',
-  $ca_port               = hiera('puppet::ca_port','8141'),
-  $client_nets           = hiera('client_nets',['127.0.0.1','::1']),
-  $ca_ttl                = '10y',
-  $daemonize             = true,
-  $enable_ca             = true,
-  $enable_master         = true,
-  $environmentpath       = $::pupmod::params::puppet_config['environmentpath'],
-  $freeze_main           = false,
-  $masterport            = '8140',
-  $puppet_confdir        = $::pupmod::params::puppet_config['confdir'],
-  $confdir               = $::pupmod::params::master_config['confdir'],
-  $codedir               = $::pupmod::params::master_config['codedir'],
-  $vardir                = $::pupmod::params::master_config['vardir'],
-  $rundir                = $::pupmod::params::master_config['rundir'],
-  $logdir                = $::pupmod::params::master_config['logdir'],
-  $use_legacy_auth_conf  = true,
-  $use_iptables          = defined('$::use_iptables') ? { true => str2bool($::use_iptables), default => hiera('use_iptables', true) },
-  $ca_status_whitelist   = [$::fqdn],
-  $ruby_load_path        = '',
-  $max_active_instances  = (to_integer($::processorcount) + 2),
-  $ssl_protocols         = [ 'TLSv1', 'TLSv1.1', 'TLSv1.2' ],
-  $ssl_cipher_suites     = [],
-  $enable_profiler       = false,
-  $admin_api_whitelist   = [$::fqdn],
-  $admin_api_mountpoint  = '/puppet-admin-api',
-  $log_to_file           = false,
-  $log_to_syslog         = true,
-  $syslog_facility       = 'LOCAL6',
-  $syslog_message_format = '%logger[%thread]: %msg',
-  $log_level             = 'WARN'
+  Simplib::IP                    $bind_address          = '0.0.0.0',
+  Simplib::IP                    $ca_bind_address       = '0.0.0.0',
+  Simplib::Port                  $ca_port               = simplib::lookup('simp_options::puppet::ca_port', { 'default_value' => 8141 }),
+  Simplib::NetList               $trusted_nets          = simplib::lookup('simp_options::trusted_nets', { 'default_value' => ['127.0.0.1','::1'] }),
+  Pattern['^\d+y$']              $ca_ttl                = '10y',
```

Comment:
This is not correct https://docs.puppet.com/puppet/latest/configuration.html#cattl